packages/common/src/hooks/usePagination/usePagination.ts

```diff
@@ -17,17 +17,27 @@ export interface PaginationState extends PaginationController {
 }
 
 export const usePagination = (initialFirst = 20): PaginationState => {
+  const searchParams = useSearchParameters();
+  const getPageFromSearchParams = () =>
+    Math.max(0, searchParams.getNumber('page') - 1);
   const [first, onChangeFirst] = useState(initialFirst);
   const [offset, onChangeOffset] = useState(0);
-  const [page, onChangePage] = useState(0);
+  const [page, onChangePage] = useState(getPageFromSearchParams());
 
   const pageRef = useRef(page);
 
   useEffect(() => {
     onChangeOffset(page * first);
     pageRef.current = page;
+    searchParams.set({ page: String(page + 1) });
   }, [first, page]);
 
+  useEffect(() => {
+    const newPage = getPageFromSearchParams();
+    onChangeOffset(newPage * first);
+    pageRef.current = newPage;
+  }, [location]);
+
```

Comment:
If for example I click on the navigation menu for the same page i'm on, the page param disappears. Not a big deal but seems inconsistent..

Thought something like

```
useEffect(() => {
    if (!searchParams.getNumber('page')) {
      searchParams.set({ page: String(page + 1) });
    }
  });
```

similar with the initial sort state, though I see you have a test that the initial state is empty, so maybe this is not right

Reply:
good catch! thanks